src/main/java/net/dries007/tfc/util/skills/SimpleSkill.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.util.skills;
+
+import javax.annotation.Nonnull;
+
+import net.minecraft.nbt.NBTTagCompound;
+
+import net.dries007.tfc.api.capability.player.IPlayerData;
+
+public class SimpleSkill extends Skill
+{
+    private float amount;
+
+    public SimpleSkill(IPlayerData rootSkills)
+    {
+        super(rootSkills);
+        this.amount = 0;
+    }
+
+    @Override
+    @Nonnull
+    public SkillTier getTier()
+    {
+        return SkillTier.valueOf((int) amount);
+    }
+
+    @Override
+    public float getLevel()
+    {
+        // checks >=4f for full progress bar in MASTER tier.
+        return amount >= 4f ? 1.0F : amount % 1.0F;
+    }
+
+    @Override
+    public void setTotalLevel(double value)
+    {
+        amount = (float) value * 4f;
```

Comment:
This should do the same verification check that `add` does - don't allow a value outside [0, 1]